lib/internal/fs/streams.js

```diff
@@ -79,6 +113,29 @@ function close(stream, err, cb) {
   }
 }
 
+function importFd(stream, options) {
+  stream.fd = null;
+  if (options.fd) {
+    if (typeof options.fd === 'number') {
+      // When fd is a raw descriptor, we must keep our fingers crossed
+      // that the descriptor won't get closed, or worse, replaced with
+      // another one
+      // https://github.com/nodejs/node/issues/35862
+      stream.fd = options.fd;
+    } else {
```

Comment:
I think you should add tests to ensure it throws if `options.fd` is a `string`, a `BigInt`, a `Symbol`, etc.
```suggestion
    } else if (typeof options.fd.fd === 'number') {
```

Reply:
I will restore the `instanceOf` in the else, it is not such a big deal
Whoever transformed the test to async function, forgot to call it 😄 
I will fix it

Reply:
@aduh95 who is supposed to squash before merge? the author or the commiter?

Reply:
The author will, unless you comment to request them to land it as separate commits.

Reply:
The author? So I should squash? Or I should simply comment if the committer is should not squash?

Reply:
Sorry, I meant the committer! It's OK if you squash them yourself, it will be squashed on landing anyway unless stated otherwise.